package sp.db;

import "db_record_msg.proto";

message visited_url
{
 required string url = 1;
 required int32 hits = 2; /* url hits for this query. */
 optional string title = 3; /* url title. */
 optional string summary = 4; /* snippet summary. */
 optional uint32 url_date = 5; /* URL data date. */
 optional string url_lang = 6; /* URL language. */
 optional uint32 rec_date = 7; /* record date, for recording activity. */
}

message visited_urls
{
 repeated visited_url vurl = 1;
}

message related_queries
{
 repeated related_query rquery = 1;
}

message related_query
{
 required uint32 radius = 1;            /* similarity radius to the original query. */
 required string query = 2;             /* query (may be hashed). */
 required uint32 query_hits = 3;        /* number of query hits. */
 required visited_urls vurls = 4;       /* visited urls for this query. */
}

extend sp.db.record
{
 optional related_queries queries = 20; /* original queries */
}
